src/pools/subClasses/corePool.ts

```diff
@@ -0,0 +1,107 @@
+import {curve} from "../../curve.js";
+import {GaugePool, IGaugePool} from "./gaugePool";
+
+export interface ICorePool {
+    id: string;
+    name: string;
+    fullName: string;
+    symbol: string;
+    referenceAsset: string;
+    address: string;
+    lpToken: string;
+    gauge: IGaugePool;
+    zap: string | null;
+    sRewardContract: string | null;
+    rewardContract: string | null;
+    implementation: string | null;
+    isPlain: boolean;
+    isLending: boolean;
+    isMeta: boolean;
+    isCrypto: boolean;
+    isFake: boolean;
+    isFactory: boolean;
+    isMetaFactory: boolean;
+    isNg: boolean;
+    isLlamma: boolean;
+    basePool: string;
+    metaCoinIdx: number;
+    underlyingCoins: string[];
+    wrappedCoins: string[];
+    underlyingCoinAddresses: string[];
+    wrappedCoinAddresses: string[];
+    underlyingDecimals: number[];
+    wrappedDecimals: number[];
+    useLending: boolean[];
+    inApi: boolean;
+}
+
+export class CorePool implements ICorePool {
+    id: string;
+    name: string;
+    fullName: string;
+    symbol: string;
+    referenceAsset: string;
+    address: string;
+    lpToken: string;
+    gauge: IGaugePool;
+    zap: string | null;
+    sRewardContract: string | null;
+    rewardContract: string | null;
+    implementation: string | null;
+    isPlain: boolean;
+    isLending: boolean;
+    isMeta: boolean;
+    isCrypto: boolean;
+    isFake: boolean;
+    isFactory: boolean;
+    isMetaFactory: boolean;
+    isNg: boolean;
+    isLlamma: boolean;
+    basePool: string;
+    metaCoinIdx: number;
+    underlyingCoins: string[];
+    wrappedCoins: string[];
+    underlyingCoinAddresses: string[];
+    wrappedCoinAddresses: string[];
+    underlyingDecimals: number[];
+    wrappedDecimals: number[];
+    useLending: boolean[];
+    inApi: boolean;
+
+    constructor(id: string) {
+        const poolData = curve.getPoolsData()[id];
```

Comment:
Can we pass the required data to the constructor, please?
```suggestion
    constructor(readonly poolData: ICorePool) {
        const id = poolData.id;
```

Reply:
Can we do this in a future reactor?

Reply:
I think this one must be done since we have data arg in master already.